wallet-new/src/Cardano/Wallet/Kernel/Keystore.hs

```diff
@@ -0,0 +1,239 @@
+{-- | An opaque handle to a keystore, used to read and write 'EncryptedSecretKey'
+      from/to disk.
+
+    NOTE: This module aims to provide a stable interface with a concrete
+    implementation concealed by the user of this module. The internal operations
+    are currently quite inefficient, as they have to work around the legacy
+    'UserSecret' storage.
+--}
+
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase                 #-}
+
+module Cardano.Wallet.Kernel.Keystore (
+      Keystore -- opaque
+      -- * Constructing a keystore
+    , bracketKeystore
+    , bracketLegacyKeystore
+    -- * Destroying a keystore (something you rarely should do)
+    , releaseAndDestroyKeystore
```

Comment:
I think this function doesn't work so well anymore, it doesn't make sense to have destruction from _within_ the bracket, and from _outside_ the bracket we have no keystore to restore.

Reply:
I think the correct API here is to have the bracket functions themselves take an argument telling it whether or not the file should be deleted on release.